redisinsight/api/src/modules/database-recommendation/providers/database-recommendation.provider.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { DatabaseRecommendationEntity }
+  from 'src/modules/database-recommendation/entities/database-recommendation.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { plainToClass } from 'class-transformer';
+import { RecommendationDto, RecommendationsDto } from 'src/modules/database-recommendation/dto';
+import { ClientMetadata } from 'src/common/models';
+
+@Injectable()
+export class DatabaseRecommendationProvider {
+  private readonly logger = new Logger('DatabaseRecommendationProvider');
+
+  constructor(
+    @InjectRepository(DatabaseRecommendationEntity)
+    private readonly repository: Repository<DatabaseRecommendationEntity>,
+  ) {}
+
+  /**
+   * Save entire entity
+   * @param databaseId
+   * @param recommendationName
+   */
+  async create(databaseId: string, recommendationName: string): Promise<RecommendationDto> {
+    this.logger.log('Creating database recommendation');
+    return await this.repository.save(
```

Comment:
In fact DatabaseRecommendationEntity will be returned. Strange...

Reply:
Also usually there is no need for await when you return Promise in the async function

Reply:
Got it